### DISCUSSION TITLE
VSCode: Set a main/primary doc for render/preview

### DISCUSSION THREAD

**Discussion by stefanbringuier** [quarto-dev/quarto-cli#2590] [Q&A] [answered]
Is there a way to set a main/primary `.qmd` file to be used when using the quarto render/preview function in VSCode? For example, say I have the main document, `presentation.qmd` , which has the following:

```markdown
---
title: "A Presentation"
date-format: short
bibliography: _references/references.bib

format:
    revealjs:
         output: true
         reference-location: document

from: markdown+emoji
jupyter: julia-1.8
---

{{< slides/_intro.qmd >}}

{{< slides/_details.qmd >}}

...

```

If I'm editing `slides/_details.qmd` and want to preview the presentation, how can I do this without having to switch back[^a] to `presentation.qmd` ? For example, the [LaTeX Workshop extension for VSCode](https://marketplace.visualstudio.com/items?itemName=James-Yu.latex-workshop) has a setting to do this.

[^a]: If I preview/render `slides/_details.qmd` it produces a single default html page, which is not what I want.

**Comment by mcanouil** [comment, marked as answer]
For now it's not possible and I believe that is also the case for RStudio.

This feature would require to perform static code analysis to guess which document is the parent and would also require to watch all children documents which then might become a performance issue.

**Reply by stefanbringuier** [reply]
> For now it's not possible and I believe that is also the case for RStudio.
> 
> This feature would require to perform static code analysis to guess which document is the parent and would also require to watch all children documents which then might become a performance issue.

I see, thanks for the insight.

**Reply by mcanouil** [reply]
It's something that could be interesting to think more about and test in the future.
